### PR TITLE
Add tools

### LINKED ISSUE #1705: [Tool request] leena
https://github.com/mmicu/leena

### PR DESCRIPTION
**Pull request by Edu4rdSHL**
https://github.com/Edu4rdSHL/blackarch/commit/dcb391b08f106b558c893b40126675bc483e31c2 really fix #1705 and not #1706 as i commited.

### CODE REVIEW

**Comment by Edu4rdSHL** [comment]
https://github.com/BlackArch/blackarch/pull/2072/commits/d572b5b79fdc460d41614ccb6880abf80cc066aa also fixes  #1324

File: expose/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+depends=('nodejs' 'npm' 'clang' 'gnuplot' 'make' 'python2' 'mitmproxy')
+makedepends=('git')
+source=("git+https://github.com/ExpoSEJS/$_pkgname.git")
+sha512sums=('SKIP')
+
+pkgver() {
+  cd "$_pkgname"
+
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
+
```
Comment (FFY00) [inline expose/PKGBUILD]: No need for a new line here.

File: expose/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+groups=('blackarch' 'blackarch-automation' 'blackarch-binary' 'blackarch-reversing')
+url='https://github.com/ExpoSEJS/ExpoSE'
+license=('MITM')
+depends=('nodejs' 'npm' 'clang' 'gnuplot' 'make' 'python2' 'mitmproxy')
+makedepends=('git')
+source=("git+https://github.com/ExpoSEJS/$_pkgname.git")
+sha512sums=('SKIP')
+
+pkgver() {
+  cd "$_pkgname"
```
Comment (FFY00) [inline expose/PKGBUILD]: No need for quotes, `$_pkgname` doesn't contain spaces.

File: expose/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  npm install
+}
+
+package() {
+  cd "$_pkgname"
+
+  install -dm 755 "$pkgdir/usr/bin"
+  install -dm 755 "$pkgdir/usr/share/$pkgname"
+
+  install -Dm 644 -t "$pkgdir/usr/share/doc/$pkgname" README.md
```
Comment (FFY00) [inline expose/PKGBUILD]: No need for `-t`. Let's keep this simple and easy to read.

File: packages/apkid/PKGBUILD [hunk 4]
```diff
@@ -1,43 +1,32 @@
...
-
-  python2 setup.py build
-}
-
 package() {
-  cd "$srcdir/APKiD"
+  cd "$_pkgname"
 
-  python2 setup.py install --root="$pkgdir" --optimize=1
+  python setup.py build install --root="$pkgdir" --optimize=1
```
Comment (FFY00) [inline packages/apkid/PKGBUILD]: No! `build` should be run in `build()`!

File: packages/backdoor-apk/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  install -Dm 644 LICENSE "$pkgdir/usr/share/licenses/$pkgname/LICENSE"
+
+  cp -a --no-preserve=ownership "$pkgname"/* "$pkgdir/usr/share/$pkgname"
+
+  cat > "$pkgdir/usr/bin/$pkgname" << EOF
+#!/bin/sh
+cd /usr/share/$pkgname
+exec sh $pkgname.sh "\${@}"
+EOF
+  chmod a+x "$pkgdir/usr/bin/$pkgname"
```
Comment (FFY00) [inline packages/backdoor-apk/PKGBUILD]: Add a new line before running chmod to make this cleaner.

File: packages/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+# See COPYING for license details.
+
+pkgname=leena
+pkgver=2.5119f56
+pkgrel=1
+pkgdesc='Symbolic execution engine for JavaScript'
+arch=('any')
+groups=('blackarch' 'blackarch-automation')
+url='https://github.com/mmicu/leena'
+license=('GPL')
```
Comment (FFY00) [inline packages/PKGBUILD]: `GLP3`

File: expose/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+pkgname=expose
+_pkgname=ExpoSE
+pkgver=732.e47eb2e
+pkgrel=1
+pkgdesc='A Dynamic Symbolic Execution (DSE) engine for JavaScript'
+arch=('any')
+groups=('blackarch' 'blackarch-automation' 'blackarch-binary' 'blackarch-reversing')
+url='https://github.com/ExpoSEJS/ExpoSE'
+license=('MITM')
```
Comment (FFY00) [inline expose/PKGBUILD]: `MITM`? Never heard of that license :thinking:.

File: expose/PKGBUILD (hunk 3 above)
Comment (Edu4rdSHL) [reply expose/PKGBUILD]: I don't understand what do you mean.

**Comment by Edu4rdSHL** [comment]
@mrsmith0x00 I fixed that.

File: packages/PKGBUILD (hunk 6 above)
Comment (Edu4rdSHL) [reply packages/PKGBUILD]: Fixed, also the location was fixed.

**Comment by noptrix** [comment]
thanks a lot @Edu4rdSHL :)

**Pull request merged**